client/helpers/scenes.ts

```diff
@@ -187,199 +179,319 @@ export async function createSceneAndSceneItems({
+    if (verbose)
+      onMsg(
+        `Creating ${createItemReqs.length} scene items at depth ${depth}...`
+      );
+    // Await is used intentionally to defer loop iteration
+    // until all scene items have been created at each depth.
+    const {
+      batchOps: queuedBatchOps,
+      batchErrors: queuedBatchErrors,
+      itemErrors: batchItemErrors,
+    } =
+      // eslint-disable-next-line no-await-in-loop
+      await createSceneItemBatch({
+        client,
+        createSceneItemReqs: createItemReqs,
+        failFast,
+        onProgress,
+        limit,
+        sceneId,
+        polling,
+      });
+
+    resultQueuedOps = resultQueuedOps.concat(queuedBatchOps);
+    resultBatchErrors = resultBatchErrors.concat(queuedBatchErrors);
+
+    if (batchItemErrors.length) {
+      if (verbose)
+        onMsg(
+          `WARNING: ${batchItemErrors.length} scene item creation errors at depth ${depth}.`
+        );
+      resultItemErrors = resultItemErrors.concat(batchItemErrors);
+      if (failFast) {
+        createFailed = true;
+        break;
+      } else {
+        // evaluate item errors and generate retry list
+        const retryErrors: SceneItemError[] = batchItemErrors.filter((v) =>
+          isPartNotFoundError(v.res)
+        );
+        const retries: CreateSceneItemRequest[] =
+          retryErrors.map<CreateSceneItemRequest>((itemError) => {
+            const item = itemError.req;
+            return {
+              data: {
+                type: 'scene-item',
+                attributes: {
+                  ...item.attributes,
+                  metadata: {
+                    ...item.attributes.metadata,
+                    [SceneItemSystemMetadata.IsMissingGeometry]:
+                      toMetadataOrUndefined('1'),
+                    [SceneItemSystemMetadata.MissingGeometrySetId]:
+                      toMetadataOrUndefined(
+                        item.relationships.source?.data.id,
+                        item.relationships.source?.data.type === 'geometry-set'
+                      ),
+                    [SceneItemSystemMetadata.MissingPartRevisionId]:
+                      toMetadataOrUndefined(
+                        item.relationships.source?.data.id,
+                        item.relationships.source?.data.type === 'part-revision'
+                      ),
+                    [SceneItemSystemMetadata.MissingSuppliedPartId]:
+                      toMetadataOrUndefined(
+                        item.attributes.source?.suppliedPartId
+                      ),
+                    [SceneItemSystemMetadata.MissingSuppliedPartRevisionId]:
+                      toMetadataOrUndefined(
+                        item.attributes.source?.suppliedRevisionId
+                      ),
+                  },
+                  source: undefined,
+                },
+                relationships: {
+                  ...item.relationships,
+                  source: undefined,
```

Comment:
Why do we need to copy the `item` here? Is it an option to just return it? Why is the `source` set to undefined on a retry?